### ISSUE TITLE
Parsing error: Unexpected character '#' for ES11

### ISSUE THREAD

**Issue by JavierRSobrino** [microsoft/vscode-eslint#934]
With this source code on VSCode:
```

class P {
	#p = 0;
	ppp() {
	  return this.#p;
	}
}
```

It does detect an error that shouldn't be so in ES11:

`Parsing error: Unexpected character '#' `

My `.eslintrc.json` is:

```
{
    "env": {
        "browser": true,
        "node": true,
        "commonjs": true,
        "es2020": true
        //"es6": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 11
    },
    "rules": {
    }
}
```
I have eslint installed globabally:

```
$ eslint --version
v6.8.0
```

**Comment by dbaeumer** [comment]
@JavierRSobrino  does the validation happen correctly when you use eslint in the terminal. If this is the case can you please provide me with a GitHub repository I can clone that demos what you are experiencing.
